transport/spi/src/main/kotlin/EndpointComponent.kt

```diff
@@ -65,21 +67,38 @@ abstract class EndpointComponent<T : Any>(
         private const val KEEP_ALIVE_FILE_NAME = "keep-alive.lock"
         private val logger: Logger = LoggerFactory.getLogger(EndpointComponent::class.java)
 
+        /**
+         * Get the keep-alive file in the user's home directory.
+         */
+        private fun getKeepAliveFile(): File = File(System.getProperty("user.home"), KEEP_ALIVE_FILE_NAME)
+
         /**
          * Sleep while a keep-alive file exists. This is helpful in case a user terminal session is opened in the
          * Kubernetes pod, and the pod should not terminate immediately after its work is done, giving the user
          * arbitrary extra time for detailed problem analysis directly in the pod's execution environment.
          */
         suspend fun sleepWhileKeepAliveFileExists() {
-            val file = File(System.getProperty("user.home"), KEEP_ALIVE_FILE_NAME)
-            while (file.exists()) {
+            val file = getKeepAliveFile()
+            while (getKeepAliveFile().exists()) {
```

Comment:
Nit: This should probably be `while (file.exists())` after extracting the variable.